### PR TITLE
68 factory and strategy changes

### LINKED ISSUE #68: Refactor to address Factory and Strategy Design Patters
Since our entities don't make use on an inheritance structure, it is acceptable to remove our factories for the time being. 
Since Create Task and Create Event controllers both use the same code to convert day names to int indexes, we could refactor this into a method in a DayConverter class, working with the strategy design pattern.

### PR DESCRIPTION
**Pull request by DanielLivshits**
## Modification: Removing Factories and Adding Strategies 

Description:  
Removed the Factory classes due to redundancy (since our entities don't involve any inheritance structure, their implementation was unneccessary). 
Added a package for strategies, including the DayToIndexConverter, which converts the input (the name of a day) to an index corresponding to that day.

### CODE REVIEW

File: .idea/misc.xml [hunk 1]
```diff
@@ -3,5 +3,9 @@
   <component name="JavadocGenerationManager">
     <option name="OUTPUT_DIRECTORY" value="$PROJECT_DIR$/../Javadoc" />
   </component>
+  <component name="PWA">
+    <option name="enabled" value="true" />
+    <option name="wasEnabledAtLeastOnce" value="true" />
+  </component>
```
Comment (Fard-Faru) [inline .idea/misc.xml:9]: We keeping this?

**Review by Fard-Faru — CHANGES_REQUESTED** [review]

File: .idea/misc.xml (hunk 1 above)
Comment (DanielLivshits) [reply .idea/misc.xml:9]: it keeps crawling back huh

**Review by Fard-Faru — APPROVED** [review]
Thank you for your contribution, looks good Daniel

**Review by missionStrawberry — APPROVED** [review]
This was a good idea, Daniel! Thanks for figuring it out.

**Review by scissorkhizer — APPROVED** [review]
WOWWW great job Daniel! With this we can directly address the feedback left by our TA on the use of factories.

**Pull request merged**